### PR TITLE
Fix crash when auth.gog.com is down

### LINKED ISSUE #354: minigalaxy couldn't start because auth.gog.com is down
Hello,

Tried to start minigalaxy and got the following Python stack trace:
```
$ MG_DEBUG=1 ./bin/minigalaxy 
Traceback (most recent call last):
  File "/home/laurent/Devel/github/minigalaxy/./bin/minigalaxy", line 64, in <module>
    main()
  File "/home/laurent/Devel/github/minigalaxy/./bin/minigalaxy", line 58, in main
    window = Window(APPLICATION_NAME)
  File "/home/laurent/Devel/github/minigalaxy/minigalaxy/ui/window.py", line 70, in __init__
    self.__authenticate()
  File "/home/laurent/Devel/github/minigalaxy/minigalaxy/ui/window.py", line 172, in __authenticate
    authenticated = self.api.authenticate(refresh_token=token, login_code=url)
  File "/home/laurent/Devel/github/minigalaxy/minigalaxy/api.py", line 29, in authenticate
    return self.__refresh_token(refresh_token)
  File "/home/laurent/Devel/github/minigalaxy/minigalaxy/api.py", line 43, in __refresh_token
    response_token = self.__get_refresh_token(params)
  File "/home/laurent/Devel/github/minigalaxy/minigalaxy/api.py", line 62, in __get_refresh_token
    response_params = response.json()
  File "/usr/lib/python3.9/site-packages/requests/models.py", line 900, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib64/python3.9/site-packages/simplejson/__init__.py", line 525, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.9/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/lib64/python3.9/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

It seems that auth.gog.com is currently down, I've random 503/504 errors when I try to access https://auth.gog.com/ in my browser.

Having auth.gog.com down shouldn't prevent minigalaxy from starting. I should still be able to play installed games.

### PR DESCRIPTION
**Pull request by lmeunier**
Fix #354

### CODE REVIEW

**Comment by lmeunier** [comment]
This is a naive approach to fix the issue with auth.gog.com. Maybe you could fix this issue with a way better solution than mine.

**Comment by makson96** [comment]
Hi,
Thanks for your support. I think that better place to address this issue is in line 61 in api.py. For example:
```
        try:
            response_params = response.json()
        except Exception as e:
            print(e)
            response_params = {}
```
This will help to address this exception if other function enters this code here.

**Comment by lmeunier** [comment]
I'm not sure about moving the try/except in api.py instead of window.py. Doing this will still display the Login dialog, but with an error message. And the only way to hide/close this dialog is the close button, which will trigger `Gtk.main_quit()` in https://github.com/sharkwouter/minigalaxy/tree/master/minigalaxy/ui/window.py#L180 and close the app.

File: minigalaxy/ui/window.py [hunk 1]
```diff
@@ -67,7 +67,11 @@ def __init__(self, name="Minigalaxy"):
             os.makedirs(THUMBNAIL_DIR, mode=0o755)
 
         # Interact with the API
-        self.__authenticate()
+        try:
+            self.__authenticate()
+        except Exception as e:
+            print(e)
+            self.offline = True
         self.HeaderBar.set_subtitle(self.api.get_user_info())
```
Comment (sharkwouter) [inline minigalaxy/ui/window.py:75]: Wouldn't this cause an exception in this case?

**Comment by sharkwouter** [comment]
Thanks for the PR. I don't think window.py is a bad place to have this check, since that is where we keep track of the offline mode. It would be good if we had a test for this, though, to made sure there are no other exceptions when the authentication fails.

File: minigalaxy/ui/window.py (hunk 1 above)
Comment (lmeunier) [reply minigalaxy/ui/window.py:75]: No, I don't think. From what I saw, the Login dialog stay open as long as the authentication is not successful.

**Comment by sharkwouter** [comment]
I'm just going to merge this, because I've not seen any problems with it and it looks good. Thanks for the contribution!

**Pull request merged**